clouddriver-oracle-bmcs/src/main/groovy/com/netflix/spinnaker/clouddriver/oraclebmcs/provider/view/OracleBMCSInstanceProvider.groovy

```diff
@@ -52,7 +52,7 @@ class OracleBMCSInstanceProvider implements InstanceProvider<OracleBMCSInstance>
       log.warn("There should be at most one identifier!")
     }
     Set<OracleBMCSInstance> instances = loadInstances(identifiers)
-    return instances.first()
+    return instances?.isEmpty() ? null: instances.first()
```

Comment:
Would be more idiomatic to use: `return instances?.first()`

That would also avoid an NPE when `instances` is `null` (which I'm guessing it can't be, but still seems safer that way).